vcd/resource_vcd_lb_service_monitor.go

```diff
@@ -0,0 +1,339 @@
+	vcdClient := meta.(*VCDClient)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	readLBMonitor, err := edgeGateway.ReadLBServiceMonitor(&types.LBMonitor{ID: d.Id()})
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("unable to find load balancer service monitor with ID %s: %s", d.Id(), err)
+	}
+
+	return flattenLBMonitor(d, readLBMonitor)
+}
+
+func resourceVcdLbServiceMonitorUpdate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	updateLBMonitorConfig, err := expandLBMonitor(d)
+	if err != nil {
+		return fmt.Errorf("could not expand monitor for update: %s", err)
+	}
+
+	updatedLBMonitor, err := edgeGateway.UpdateLBServiceMonitor(updateLBMonitorConfig)
+	if err != nil {
+		return fmt.Errorf("unable to update load balancer service monitor with ID %s: %s", d.Id(), err)
+	}
+
+	if err := flattenLBMonitor(d, updatedLBMonitor); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceVcdLbServiceMonitorDelete(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	err = edgeGateway.DeleteLBServiceMonitor(&types.LBMonitor{ID: d.Id()})
+	if err != nil {
+		return fmt.Errorf("error deleting load balancer service monitor: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+// resourceVcdLbServiceMonitorImport expects dot formatted path to Load Balancer Service Monitor
+// i.e. my-org.my-org-vdc.my-edge-gw.existing-sm
+func resourceVcdLbServiceMonitorImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+
+	resourceURI := strings.Split(d.Id(), ".")
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("resource name must be specified in such way my-org.my-org-vdc.my-edge-gw.existing-sm")
+	}
+	orgName, vdcName, edgeName, monitorName := resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3]
+
+	vcdClient := meta.(*VCDClient)
+	edgeGateway, err := vcdClient.GetEdgeGateway(orgName, vdcName, edgeName)
+	if err != nil {
+		return nil, fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	readLBMonitor, err := edgeGateway.ReadLBServiceMonitor(&types.LBMonitor{Name: monitorName})
+	if err != nil {
+		return []*schema.ResourceData{}, fmt.Errorf("unable to find load balancer service monitor with ID %s: %s", d.Id(), err)
+	}
+
+	d.Set("org", orgName)
+	d.Set("vdc", vdcName)
+	d.Set("edge_gateway", edgeName)
+	d.Set("name", monitorName)
+
+	d.SetId(readLBMonitor.ID)
+	return []*schema.ResourceData{d}, nil
```

Comment:
We add only this values cause, user will have to add other manually?

Reply:
No. The point of import is that you lookup the object ID and it calls `terraform refresh` implicitly which populates all the other data using the `read` method as far as I've seen.

Reply:
Why not import automatically without refresh? - cause in terraform documentation I don't `refresh` mentioned as process.

Reply:
I could of course set them here as well however what I saw while investigating other providers is they're setting only defaults (which are not populated by read/refresh. However I aim to read all fields via import.

As far as the refresh behaviour -  you can't change that behaviour. That's how import looks like (You will see it is refreshing):
```bash
$ terraform import vcd_lb_service_monitor.import my-org.my-org-vdc.my-edge-gw.service-monitor-1
vcd_lb_service_monitor.import: Importing from ID "my-org.my-org-vdc.my-edge-gw.service-monitor-1"...
vcd_lb_service_monitor.import: Import complete!
  Imported vcd_lb_service_monitor
vcd_lb_service_monitor.import: Refreshing state... [id=monitor-140]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```


After this operation the statefile has such fields:
```json
{
      "mode": "managed",
      "type": "vcd_lb_service_monitor",
      "name": "import",
      "provider": "provider.vcd",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "edge_gateway": "my-edge-gw",
            "expected": "HTTP/1.1",
            "extension": {
              "content-type": "application/json",
              "no-body": ""
            },
            "id": "monitor-140",
            "interval": 5,
            "max_retries": 3,
            "method": "POST",
            "name": "service-monitor-1",
            "org": "my-org",
            "receive": "OK",
            "send": "{\"key\": \"value\"}",
            "timeout": 5,
            "type": "http",
            "url": "",
            "vdc": "my-org-vdc"
          }
        }
      ]
    },
```

Reply:
ok. Tx for info